src/test/java/seedu/sudohr/model/department/DepartmentTest.java

```diff
@@ -19,6 +22,50 @@ public void asObservableList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> department.getEmployees().remove(0));
     }
 
+    @Test
+    public void execute_addEmployee_success() {
+        Department department = new DepartmentBuilder().build();
+
+        department.addEmployee(TypicalEmployees.ALICE);
+        assertTrue(department.hasEmployee(TypicalEmployees.ALICE));
+    }
+
+    @Test
+    public void execute_addDuplicateEmployee_throwsException() {
+        Department department = new DepartmentBuilder().build();
+
+        department.addEmployee(new EmployeeBuilder().build());
+
+        assertThrows(DuplicateEmployeeException.class, () -> department.addEmployee(new EmployeeBuilder().build()));
+    }
+
+    @Test
+    public void execute_addExactDuplicateEmployee_throwsException() {
+        Department department = new DepartmentBuilder().build();
+
+        department.addEmployee(TypicalEmployees.ALICE);
+
+        assertThrows(DuplicateEmployeeException.class, () -> department.addEmployee(TypicalEmployees.ALICE));
+    }
+
+    @Test
```

Comment:
May want to consider testing removal of an employee that does not exist / been removed.

or an integrated test where you remove, then add an employee again, just to make sure no error thrown in any levels.

Reply:
yessir, will work on it in my next issue for RemoveEmployeeFromDepartmentCommand